include/execution.hpp

```diff
@@ -2633,7 +2633,15 @@ namespace std::execution {
               memory_order_acquire));
 
           if (__old == nullptr) {
-            start(__shared_state->__op_state2_);
+            // the inner sender isn't running
+            if (__shared_state->__stop_source_.stop_requested()) {
+              // 1. resets __head to completion state
+              // 2. notifies waiting threads
+              // 3. propagates "stopped" signal to `out_r'`
+              __shared_state->__notify();
+            } else {
+              start(__shared_state->__op_state2_);
+            }
```

Comment:
What bug exactly does this fix?

If there was a race condition with the previous implementation then there would still be a race-condition with the new implementation (e.g. what if stop is requested on another thread immediately after `stop_requested()` returns).

This seems to be purely an optimisation that avoids starting the underlying op if stop has already been requested.

Reply:
Without this change, the first of the new test cases I added will fail, and `split` won't conform to the spec.